packages/dapp/src/stores/menu/MenuStore.tsx

```diff
@@ -0,0 +1,114 @@
+import { MenuProps, MenuState } from '@transferto/shared/src/types';
+import { create } from 'zustand';
+import { MenuKeys } from '../../const';
+
+interface DefaultMenuType {
+  openMainNavbarMenu: boolean;
+  openNavbarWalletMenu: boolean;
+  openNavbarChainsMenu: boolean;
+  openNavbarWalletSelectMenu: boolean;
+  openNavbarSubMenu: keyof typeof MenuKeys;
+  openSupportModal: boolean;
+  anchorRef: null | JSX.Element;
+}
+
+export const defaultMenu: DefaultMenuType = {
+  openMainNavbarMenu: false,
+  openNavbarWalletMenu: false,
+  openNavbarChainsMenu: false,
+  openNavbarWalletSelectMenu: false,
+  openNavbarSubMenu: 'None',
+  openSupportModal: false,
+  anchorRef: null,
+};
+
+export const useMenuStore = create<MenuState>((set, get) => ({
+  ...defaultMenu,
+  setValue: (key, value) =>
+    set(() => ({
+      [key]: value,
+    })),
+  setValues: (values) =>
+    set((state) => {
+      const updatedState: MenuProps = { ...state };
+      for (const key in values) {
+        if (Object.hasOwn(state, key)) {
+          updatedState[key] = values[key];
+        }
+      }
+      return updatedState;
+    }),
+
+  // Close ALL Navbar Menus
+  onCloseAllNavbarMenus: () => {
+    set({
+      openMainNavbarMenu: false,
+      openNavbarWalletSelectMenu: false,
+      openNavbarWalletMenu: false,
+      openNavbarChainsMenu: false,
+      openNavbarSubMenu: MenuKeys.None,
+      openSupportModal: false,
+      anchorRef: null,
+    });
+  },
+
+  // Set AnchorElement on Initialization of Navbar
+  onMenuInit: () => {
+    set({});
+  },
```

Comment:
Do we need this init function if we only set an empty object here? We can have it as default value.

Reply:
No, this is not even used and belongs to the garbage! I thank you for that find